wcfsetup/install/files/lib/data/user/avatar/UserAvatarEditor.class.php

```diff
@@ -58,11 +59,64 @@ public function deleteFiles() {
 		foreach (UserAvatar::$avatarThumbnailSizes as $size) {
 			if ($this->width < $size && $this->height < $size) break;
 			
-			@unlink($this->getLocation($size));
+			@unlink($this->getLocation($size, false));
 		}
-		@unlink($this->getLocation('resize'));
+		@unlink($this->getLocation('resize', false));
 		
 		// delete original size
-		@unlink($this->getLocation());
+		@unlink($this->getLocation(null, false));
+		
+		if ($this->hasWebP) {
+			@unlink($this->getLocation(null, true));
+		}
+	}
+	
+	/**
+	 * Creates a WebP variant of the avatar, unless it is a GIF image. If the
+	 * user uploads a WebP image, this method will create a JPEG variant as a
+	 * fallback for ancient clients.
+	 * 
+	 * Will return `true` if a variant has been created.
+	 *
+	 * @since 5.4
+	 */
+	public function createAvatarVariant(): bool {
+		if ($this->hasWebP) {
+			return false;
+		}
+		
+		if ($this->avatarExtension === "gif") {
+			// We do not touch GIFs at all.
+			return false;
+		}
+		
+		$filename = $this->getLocation();
+		$filenameWebP = $this->getLocation(null, true);
+		
+		$imageAdapter = ImageHandler::getInstance()->getAdapter();
+		$imageAdapter->loadFile($filename);
+		$image = $imageAdapter->getImage();
+		
+		$data = ["hasWebP" => 1];
+		
+		// If the uploaded avatar is already a WebP image, then create a JPEG
+		// as a fallback image and flip the image data to match the JPEG.
+		if ($this->avatarExtension === "webp") {
+			$filenameJpeg = preg_replace('~\.webp$~', '.jpeg', $filenameWebP);
+			
+			$imageAdapter->saveImageAs($image, $filenameJpeg, "jpeg", 80);
+			
+			$data = [
+				"avatarExtension" => "jpeg",
+				"fileHash" => sha1_file($filenameJpeg),
```

Comment:
Did you check this? Changing the `fileHash` should result in the code searching the avatar in a different location, breaking it.